### ISSUE TITLE
feat: gather timer become obsolete  when any interm/hypostasis received 

### ISSUE THREAD

**Issue by vdharashive** [jambonz/jambonz-feature-server#777]
if gather has timeout of 10 sec and user start speaking at 7th sec, there are some interm/hypothesis received then gather should end at 10th sec . rather it should continue getting result until is-final =true is received , in current implimentation it getting terminated at 10 sec with timeout

**Comment by davehorton** [comment]
yes, but in that case, when a partial transcript is received at 7th second, do you want it then to extend the timeout by another 10 seconds?

**Comment by vdharashive** [comment]
timeout should be killed and final transcript has to come from asr , this logic only will apply if interim result are come

**Comment by davehorton** [comment]
ok so once an interim transcript is received the timeout will be completely removed and we will wait for the final transcript.  In the case of continuous asr however that timeout will not be removed and will continue to be active
